pkg/volume/glusterfs/glusterfs.go

```diff
@@ -763,13 +763,13 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 		return nil, 0, fmt.Errorf("error [%v] when getting cluster nodes for volume %s", err, volume)
 	}
 
-	// The 'endpointname' is created in form of 'glusterfs-dynamic-<claimname>'.
+	// The 'endpointname' is created in form of 'glusterfs-dynamic-<claimuid>'.
 	// createEndpointService() checks for this 'endpoint' existence in PVC's namespace and
 	// If not found, it create an endpoint and svc using the IPs we dynamically picked at time
 	// of volume creation.
-	epServiceName := dynamicEpSvcPrefix + p.options.PVC.Name
+	epServiceName := dynamicEpSvcPrefix + string(p.options.PVC.UID)
```

Comment:
I don't think this will help solve the issue.

/cc @msau42 @saad-ali

Reply:
@dixudx I tested, and this works. I add some explain in Special notes

Reply:
Will this break backwards compatibility?

Reply:
It shouldn't because the name is stored in the PV spec, and the deletion uses the name from the PV.

Reply:
Agree @msau42